share/functions/__fish_print_pacman_packages.fish

```diff
@@ -10,17 +10,10 @@ function __fish_print_pacman_packages
 
     if not set -q _flag_installed
         set -l cache_file $xdg_cache_home/pacman
-        if test -f $cache_file
-            cat $cache_file
-            set -l age (path mtime -R -- $cache_file)
-            set -l max_age 250
-            if test $age -lt $max_age
-                return
-            end
-        end
+        __fish_cache_read $cache_file 250 && return
         __fish_cache_put $cache_file
         # prints: <package name>	Package
-        pacman -Ssq | sed -e 's/$/\t'Package'/' >$cache_file &
+        pacman -Ssq | sed -e 's/$/\t'Package'/' | tee $cache_file
```

Comment:
I think the idea was that "no completions" is better than "hanging for a while" because the latter might force the user to press ctrl-c.
I think most other users of this cache logic do the same.

But this command seems fast enough (200ms here) so this seems like a good change.
(perhaps we don't need the cache at all.. not sure)

Reply:
> I think the idea was that "no completions" is better than "hanging for a while" because the latter might force the user to press ctrl-c.

Not sure what could be the best behavior, but I'm tending to the idea of "hanging for a while". It seems to be a little more predictable and gives the user the control. Maybe there could be also a timeout to limit the maximum time for the completion to hang